lib/asciidoctor/helpers.rb

```diff
@@ -126,13 +126,23 @@ def self.uri_prefix str
   # Matches the characters in a URI to encode
   UriEncodeCharsRx = /[^\w\-.!~*';:@=+$,()\[\]]/
 
-  # Internal: Encode a String for inclusion in a URI.
-  #
-  # str - the String to URI encode
-  #
-  # Returns the String with all URI reserved characters encoded.
-  def self.uri_encode str
-    str.gsub(UriEncodeCharsRx) { $&.each_byte.map {|c| sprintf '%%%02X', c }.join }
+  # Internal: Encode a URI component String for safe inclusion in a URI.
+  #
+  # str - the URI component String to encode
+  #
+  # Returns the String with all reserved URI characters encoded (e.g., /, &, =, space, etc).
+  if RUBY_ENGINE == 'opal'
```

Comment:
If you want I can move this code in Asciidoctor.js

Reply:
Better would be to have CGI.escape implemented in Opal.

Reply:
That's the next next step :+1:

Reply:
:+1: 

No rush. I'm just playing around with some options here. Since we're changing the characters being encoded, the goal will be for it to make it into 2.0.0.